### PR TITLE
Attempt at tests

### PR DESCRIPTION
**Pull request by dmtrskv**
The test hanging is fixed here: https://github.com/NilFoundation/nil-hardhat-plugin/pull/17
**With that fix the test here passes!**

Changes:
1. Updated readme on how to run tests.
2. Fixed env example and test command.
3. Inlined deployment code (deployContract), so that it is easier to see where it hangs. (Also, it doesn't work because of 4).
4. Fixed final address calculation (ethereum-style address calculation prevents us from using hardhat's deployContract directly).

The correctness of the test can be checked by `npx hardhat test --config hardhat.config.testing.ts` which will run it against an ethereum node.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -6,9 +6,9 @@ NIL_RPC_ENDPOINT: "http://127.0.0.1:8529"
 
 # Specify the private key used for signing transactions
 # This should be a hexadecimal string corresponding to your account's private key
-PRIVATE_KEY: "41285f03e8692676bf80a98e4052a008026427a7302ca97cb06edcd60689850b"
+PRIVATE_KEY: ""
```
Comment (Rolaman) [inline .env.example:9]: Why? I think it is better to show real case in examples

**Review by Rolaman — APPROVED** [review]

File: .env.example (hunk 1 above)
Comment (dmtrskv) [reply .env.example:9]: We should put an actual private key and wallet address available on the devnet then.
And for local runs the user must create the wallet anyway.
So, I agree with you, but I'm not sure which approach would be better.

Comment (Rolaman) [reply .env.example:9]: no big deel now, lets merge 👍

Comment (ukorvl) [reply .env.example:9]: Typically env.example contains no values

**Pull request merged**